### PR TITLE
APIKey Restriction/X-Ios-Bundle-Identifier support

### PR DESCRIPTION
**Pull request by thomasvl**

### CODE REVIEW

File: Source/Objects/GTLRService.h [hunk 1]
```diff
@@ -100,6 +100,13 @@ extern NSString *const kGTLRServiceTicketParsingStartedNotification;
  */
 extern NSString *const kGTLRServiceTicketParsingStoppedNotification;
 
+/**
+ *  The header name used for to send an Applicatoin's Bundle Identifier.
```
Comment (sergiocampama) [inline Source/Objects/GTLRService.h]: remove for
Application

File: Source/Objects/GTLRService.h [hunk 2]
```diff
@@ -264,9 +271,28 @@ typedef void (^GTLRServiceTestBlock)(GTLRServiceTicket *testTicket,
 
 /**
  *  Some services require a developer key for quotas and limits.
+ *
+ *  If you have enabled the iOS API Key Restriction, you will want
+ *  to manually set the @c APIKeyRestrictionBundleID property, or
+ *  use -includeBundleIDRestrictionWithAPIKey: to set your API key
```
Comment (sergiocampama) [inline Source/Objects/GTLRService.h]: setMainBundleIDRestrictionWithAPIKey

File: Source/Objects/GTLRService.h [hunk 3]
```diff
@@ -264,9 +271,28 @@ typedef void (^GTLRServiceTestBlock)(GTLRServiceTicket *testTicket,
...
+ *  and set the restriction to the main bundle's bundle id.
  */
 @property(nonatomic, copy, nullable) NSString *APIKey;
 
+/**
+ *  The Bundle Identifier to use for the API key restriciton. This will be
```
Comment (sergiocampama) [inline Source/Objects/GTLRService.h]: restriction

File: Source/Objects/GTLRService.h [hunk 4]
```diff
@@ -441,13 +469,22 @@ typedef void (^GTLRServiceTestBlock)(GTLRServiceTicket *testTicket,
...
+ *  is done. Library-specific details will be appended.
  *
- *  @param userAgent A wire-ready use agent string.
+ *  @param userAgent A wire-ready user agent string.
  */
 - (void)setExactUserAgent:(nullable NSString *)userAgent;
 
+/**
+ *  A precise userAgent string to send on requests; no cleaning is done. When
+ *  set, requestUserAgent be exactly this, no library or system information will
```
Comment (sergiocampama) [inline Source/Objects/GTLRService.h]: *will* be exactly this

File: Source/Objects/GTLRService.h (hunk 1 above)
Comment (thomasvl) [reply Source/Objects/GTLRService.h]: done

File: Source/Objects/GTLRService.h (hunk 2 above)
Comment (thomasvl) [reply Source/Objects/GTLRService.h]: done

File: Source/Objects/GTLRService.h (hunk 3 above)
Comment (thomasvl) [reply Source/Objects/GTLRService.h]: done

**Pull request merged**